Auto submodule update: cmake - add FindBoringSSL.cmake (#152)

swiftnav-svc-jenkins opened the pull request:
Automated PR by Jenkins. If CI has passed successfully, merge away!

**cmake**  b8d744bd  ->  38cf865b
-  38cf865b  :  add FindBoringSSL.cmake (swift-nav/cmake#152)
-  b5f4e040  :  Add TEST_SRCS property and swift_add_test_srcs_target macro [BUILD-657
-  d165dd7c  :  Add official MKL cmake finder. (swift-nav/cmake#147)
-  d7a20ec5  :  Turn SonarCloud duplicate detection off for unit tests (swift-nav/cmake#150)
-  6376d609  :  Remove pvt-runner-lib from search libraries in starling target (swift-nav/cmake#148)
-  4047c088  :  Add plantuml [BUILD-537] (swift-nav/cmake#146)
-  3f59bcdc  :  Exclude unit test header files from coverage reporting (swift-nav/cmake#145)
-  ac5dc344  :  Only set sanitizer suppression file if sanitzing is enabled
-  eb24e29b  :  Don't fail on empty diffs when running clang-format
-  5f7f2cf4  :  Add a script supporting ratcheting for clang-tidy warnings
-  b2d67a69  :  Support issue suppression for runtime sanitizers
-  3aab0ecf  :  gRCP mark protobuf as system include (swift-nav/cmake#139)
-  3d6fedf6  :  Disable starling core tests when used as a submodule (swift-nav/cmake#138)
-  2adba802  :  Add find module for libuv [PSDM-175] (swift-nav/cmake#137)
-  f4d30717  :  Add find module for Auk (swift-nav/cmake#135)
-  18206e36  :  Revert "CMake file to find the Auk project when submoduling it (#133)"
-  8ac5ddda  :  CMake file to find the Auk project when submoduling it (swift-nav/cmake#133)


This pull request was created by https://jenkins.ci.swift-nav.com/job/CI%20Infra/job/submodule-update/14104/.


[BUILD-537]: https://swift-nav.atlassian.net/browse/BUILD-537?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PSDM-175]: https://swift-nav.atlassian.net/browse/PSDM-175?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

pgrgich commented on include/swiftnav/ephemeris.h:
```diff
@@ -211,7 +211,9 @@ u8 encode_ura(float ura);
 
 u8 get_ephemeris_iod(const ephemeris_t *eph);
-double get_tgd_correction(const ephemeris_t *eph, const gnss_signal_t *sid);
+s8 get_tgd_correction(const ephemeris_t *eph,
```
I know what you mean, but all the other functions in this file do it this way (and I don't see an existing alternative in libswiftnav), so I went with this for consistency. I'd be happy to change it, but it would probably be best to change all the return values in the file if we're going to do so.

seth-swiftnav replied:
I'm also in favor of any sort of enumeration for the return codes, just so we don't rely on common conventions.
What is "TDG"?

pgrgich replied:
TGD is the Total Group Delay (code hardware bias).

If we're going to go with a return code here, we should do it throughout the file, shouldn't we? It's also more Exafore code than ours, so we should get agreement on the way forward first.

I'm happy to do so, but want to know it's agreed upon before spending the time on it. What do we think?

scarcanague replied:
Let's do a bit of cleaning and go with an ENUM

pgrgich commented:
@nsirola I'm making a change here (see the last commit) to convert to a return code rather than hardcoded 0s and 1s. Can you have a look and let me know if you have any concerns?

anth-cole commented:
While I'm 100% in favour of using return codes rather than ints, in this instance I'd keep it out of this PR and do it as a separate task. I'd worry about adding this enum only for these specific functions when the rest of the LSN library is using the s8 return pattern.

swiftnav-svc-jenkins committed:
cmake - add FindBoringSSL.cmake
* add FindBoringSSL.cmake

* whitespace

* add crypto to ADDITIONAL_TARGETS

Triggered-By:	cmake	38cf865b47106e4ecc2f0a0a0322e20981470f47
Upstream-PR:	http://github.com/swift-nav/cmake/pull/152
